enums.rb

```diff
@@ -0,0 +1,221 @@
+        yield(element)
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      for i in (0...size)
+        yield(i, self[i])
+      end
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+
+      arr = []
+      my_each do |element|
+        arr << element if yield(element)
+      end
+      arr
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(arg = nil)
+    response = true
+    if block_given?
+      my_each do |element|
+        response = false unless yield(element)
+      end
+
+    elsif !arg.nil?
+      if arg.is_a? Class
+        my_each do |element|
+          response = false unless element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = false unless element.match(arg)
+        end
+
+      else
+        my_each do |element|
+          response = false unless element.eql?(arg)
+        end
+      end
+
+    else
+      response = true
+    end
+    response
+  end
+
+  def my_any?(arg = nil)
+    response = false
+    if block_given?
+      my_each do |element|
+        response = true if yield(element)
+      end
+
+    elsif !arg.nil?
+      if arg.is_a? Class
+        my_each do |element|
+          response = true if element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = true if element.match(arg)
+        end
+
+      else
+        my_each do |element|
+          response = true if element.eql?(arg)
+        end
+      end
+
+    else
+      response = true
+    end
+    response
+  end
+
+  def my_none?(arg = nil)
+    response = false
+    if block_given?
+      response = true
+      my_each do |element|
+        response = false if yield(element)
+      end
+
+    elsif !arg.nil?
+
+      if arg.is_a? Class
+        my_each do |element|
+          response = true unless element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = true unless element.match(arg)
+        end
+
+      else
+        response = true
+        my_each do |element|
+          response = false if element.eql?(arg)
+        end
+      end
+    else
+      response = true
+    end
+    response
+  end
+
+  def my_count(count = 0)
+    if block_given?
+      my_each do |element|
+        count += 1 if yield(element)
+      end
+
+    elsif !count.eql?(0)
+      count += size
+
+    else
+      count = size
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    arr = []
+
+    if proc.equal?(nil) == false
+      my_each do |element|
+        arr << proc.call(element)
+      end
+    elsif block_given?
+      my_each do |element|
+        arr << yield(element)
+      end
+    else
+      arr = to_a
+
+    end
+    to_enum
+  end
+
+  def my_inject(* args)
+    if is_a? Range
+      to_a
+      puts self
+    end
+    if args[0].equal?(:+) and args[1].nil?
+      acum = 0
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
array = (0..100).to_a

array.my_inject(:+) != array.inject(:+)
```

Reply:
Corrections were not made **inject returns** the same output Gladly check, please
![image](https://user-images.githubusercontent.com/53732843/81597326-cba76c00-9382-11ea-8f8c-9206e6826110.png)